source/frontend/src/components/maps/leaflet/Markers/SingleMarker.tsx

```diff
@@ -0,0 +1,74 @@
+import { LatLngLiteral } from 'leaflet';
+import { Marker } from 'react-leaflet';
+import { PointFeature } from 'supercluster';
+
+import { useMapStateMachine } from '@/components/common/mapFSM/MapStateMachineContext';
+import { PMBC_FullyAttributed_Feature_Properties } from '@/models/layers/parcelMapBC';
+import { PIMS_Property_Location_View } from '@/models/layers/pimsPropertyLocationView';
+
+import { getMarkerIcon, getNotOwnerMarkerIcon } from '../Layers/util';
+
+interface SinlePropertyMarkerProps {
+  pointFeature: PointFeature<PIMS_Property_Location_View | PMBC_FullyAttributed_Feature_Properties>;
+  markerPosition: LatLngLiteral;
+  isSelected: boolean;
+  isOwned: boolean;
+}
+
+const SinglePropertyMarker: React.FC<React.PropsWithChildren<SinlePropertyMarkerProps>> = ({
+  pointFeature,
+  markerPosition,
+  isSelected,
+  isOwned,
+}) => {
+  const mapMachine = useMapStateMachine();
+
+  const getIcon = () => {
+    if (isOwned) {
+      const pimsFeature = pointFeature as PointFeature<PIMS_Property_Location_View>;
```

Comment:
I wonder if we should ever do runtime validation on any of these to catch if we make changes to the layer but forget to update the interface. What thinkest thou?